secedgar/core/combo.py

```diff
@@ -0,0 +1,167 @@
+                        self.master_date_list.append(
+                            (current_year, current_quarter, lambda x: True))
+                        current_date = next_start_quarter_date
+                    else:
+                        self.master_date_list.append(
+                            (current_year, current_quarter,
+                             lambda x: date(x['date_filed']) <= self.end_date))
+                        current_date = self.end_date
+                else:
+                    while current_date <= self.end_date:
+                        self.daily_date_list.append(current_date)
+                        current_date += timedelta(days=1)
+
+    def get_urls(self):
+        """Get all urls between ``start_date`` and ``end_date``."""
+
+        # Use functools.reduce for speed
+        # see https://stackoverflow.com/questions/10461531/merge-and-sum-of-two-dictionaries
+        def reducer(accumulator, dictionary):
+            for key, value in dictionary.items():
+                accumulator[key] = accumulator.get(key, []) + value
+            return accumulator
+
+        list_of_dicts = []
+        for (year, quarter, f) in self.master_date_list:
+            self.master.year = year
+            self.master.quarter = quarter
+            self.master.entry_filter = lambda x: f(x) and self.entry_filter(x)
+            list_of_dicts.append(self.master.get_urls())
+
+        for d in self.daily_date_list:
+            self.daily.date = d
+            try:
+                list_of_dicts.append(self.daily.get_urls())
+            except EDGARQueryError:
+                pass
+
+        complete_dictionary = reduce(reducer, list_of_dicts, {})
+        return complete_dictionary
+
+    def save(self,
+             directory,
+             dir_pattern=None,
+             file_pattern="{accession_number}",
+             download_all=False,
+             daily_date_format="%Y%m%d"):
+        """Save all filings between ``start_date`` and ``end_date``.
+
+        Only filings that satisfy args given at initialization will
+        be saved.
+
+        Args:
+            directory (str): Directory where filings should be stored.
+            dir_pattern (str, optional): Format string for subdirectories. Defaults to None.
+            file_pattern (str, optional): Format string for files. Defaults to "{accession_number}".
+            download_all (bool, optional): Type of downloading system, if true downloads
+                all data for each day, if false downloads each file in index.
+                Defaults to False.
+            daily_date_format (str, optional): Format string to use for the `{date}` pattern.
+                Defaults to "%Y%m%d".
+        """
+        for (year, quarter, f) in self.master_date_list:
+            self.master.year = year
+            self.master.quarter = quarter
+            self.master.entry_filter = lambda x: f(x) and self.entry_filter(x)
```

Comment:
@reteps this is causing an error. Try running the code below:

```python
from datetime import date
import secedgar as sec

combo = sec.ComboFilings(start_date=date(2020, 1, 1), end_date=date(2020, 5, 4), entry_filter=lambda f: f.form_type.lower() == '10-k')

combo.save('temp')
```

This produces the following error:

Traceback (most recent call last): File "/Users/jacklm/...", line 19, in combo.save('temp') File "/Users/jacklm/...", line 151, in save self.master.entry_filter = lambda x: f(x) and self.entry_filter(x) AttributeError: can't set attribute

Reply:
Either a setter on `_index.py` needs to be created or you need `self.master._entry_filter`. I would prefer we add the setter.

Source: _index L29/39
```python3
    def __init__(self, client=None, entry_filter=None, **kwargs):
        self._entry_filter = entry_filter

    @property
    def entry_filter(self):
        """A boolean function to be tested on each listing entry.

        This is tested regardless of download method.
        """
        return self._entry_filter
```

Source: `QuarterlyFilings.py` line 28
```python3
        super().__init__(client=client, entry_filter=entry_filter, **kwargs)
```

Reply:
@jackmoody11 Ok, should work now